features/lwipstack/LWIPStack.cpp

```diff
@@ -488,12 +488,18 @@ nsapi_size_or_error_t LWIP::socket_sendto(nsapi_socket_t handle, const SocketAdd
 {
     struct mbed_lwip_socket *s = (struct mbed_lwip_socket *)handle;
     ip_addr_t ip_addr;
+    nsapi_addr_t interface_addr;
 
     nsapi_addr_t addr = address.get_addr();
     if (!convert_mbed_addr_to_lwip(&ip_addr, &addr)) {
         return NSAPI_ERROR_PARAMETER;
     }
 
+    convert_lwip_addr_to_mbed(&interface_addr, get_ip_addr(true, &default_interface->netif));
+    if (addr.version != interface_addr.version) {
```

Comment:
Dual-mode is possible - that fails sending to IPv4 if you're dual IPv4 and IPv6 with IPv6 preferred.

Not to mention the issue of multiple potential interfaces - that blocks IPv6 on the IPv6 interface if your default is IPv4.

To get a good answer here, I think you'd have to get the error from lwIP. Seems like lwIP is probably generating an error internally, but it gets lost. `netconn_send` doesn't extract the `err` value from the `msg` that `netconn_do_send` fills in?

Reply:
Fixed dualstack issue as Kimmo suggested.
Checked  that if both ip4/6 enabled netconn returns ERR_OK because  ip_route always finds the netif.